libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -277,4 +280,26 @@ public void onServiceDisconnected(ComponentName className) {
       serviceBound = false;
     }
   };
+
+  boolean checkLocationPermission() {
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      return true;
+    } else {
+      permissionBackoff();
+      return false;
+    }
+  }
+
+  private void permissionBackoff() {
+    PermissionCheckRunnable permissionCheckRunnable = obtainPermissionCheckRunnable();
+    permissionCheckRunnable.run();
```

Comment:
I think that if we call `permissionCheckRunnable.run()` here the intervals will be reset (losing the exponential backoff behavior), not completely sure though we should test it.
Related 👉 https://github.com/mapbox/mapbox-events-android/pull/12/files#r155230708

Reply:
Tested and works.